openwisp_monitoring/device/api/views.py

```diff
@@ -248,5 +281,40 @@ def _create_clients_graph(self, metric):
         graph.full_clean()
         graph.save()
 
+    def _get_field_name(self, key):
+        field_dict = {'load': 'load_5', 'memory': 'free_memory'}
+        try:
+            return field_dict[key]
+        except KeyError:
+            return key
+
+    def _create_resources_graph(self, metric, resource):
+        if resource not in monitoring_settings.AUTO_GRAPHS:
+            return
+        graph = Graph(metric=metric, configuration=resource)
+        graph.full_clean()
+        graph.save()
+
+    def resources_metric_write(
```

Comment:
This function is becoming a monster :grin:. I propose removing it and `_get_field_name` too because they are complicating the readability too much only trying to make the code less repetitive.

Once removed, duplicate the code that gets_or_create the metric, write it and create the graph if needed.

E.g.

```

if 'resources' in data:
	 if 'load' in data['resources']:
		...calculate value and extra_values...
		... Metric.get_or_create() ...
		... metric.write ...
		... if created: create graphs and thresholds ...
	if 'disk_usage' in data['resources']:
                ... same ...
	if 'memory' in data['resources']:
                ... same ...
```

Then we can refactor once we find similarities in the algorithm